### PR TITLE
Removed dom-scheduler as hard-dependency (closes #5)

### LINKED ISSUE #5: Remove scheduler as a hard dependency

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by etiennesegonzac** [comment]
looks like this needs rebasing :)

File: test/karma.conf [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function(config) {
 
     // list of files / patterns to load in the browser
     files: [
-      'lib/dom-scheduler.js',
+      'dom-scheduler.js',
```
Comment (etiennesegonzac) [inline test/karma.conf]: should be `scheduled-list.js` no?

File: scheduled-list.js [hunk 2]
```diff
@@ -522,13 +556,14 @@
...
 
   /* Internals
      ========= */
 
-  var debug = false;
+  var debug2 = false;
   function log(str) {
-    if (!debug) {
+    if (!debug2) {
       return;
```
Comment (etiennesegonzac) [inline scheduled-list.js]: let's switch `log()` instances to `debug()` and get rid of this part :)
We just need a way to early return in `debugViewport` maybe
`if (debug.name !== 'log') { return; }`

File: examples/sections/bower_components/dom-scheduler/bower_components/fast-list/index.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
     <link rel="stylesheet" href="css/ul.css">
     <link rel="stylesheet" href="css/app.css">
 
-    <script src="bower_components/dom-scheduler/lib/dom-scheduler.js" defer></script>
+    <!-- <script src="bower_components/dom-scheduler/lib/dom-scheduler.js" defer></script> -->
     <script type="text/javascript" src="lib/scheduled-list.js" async></script>
```
Comment (etiennesegonzac) [inline examples/sections/bower_components/dom-scheduler/bower_components/fast-list/index.html]: probably don't need to change this :)

File: examples/sections/bower_components/dom-scheduler/bower_components/fast-list/js/page.js [hunk 4]
```diff
@@ -9,7 +12,8 @@
     var listContainer = document.querySelector('section');
 
     var source = new BaconSource();
-    var list = new ScheduledList(listContainer, source, scheduler);
+    var list = new ScheduledList(listContainer, source);
+    var scheduler = ScheduledList.scheduler;
```
Comment (etiennesegonzac) [inline examples/sections/bower_components/dom-scheduler/bower_components/fast-list/js/page.js]: and this :)

**Comment by etiennesegonzac** [comment]
I know it's just symbolic but this commit is also good opportunity to update the "authors" array of the root `bower.json` :)

File: examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js [hunk 5]
```diff
@@ -9,7 +12,8 @@
     var listContainer = document.querySelector('section');
 
     var source = new BaconSource();
-    var list = new ScheduledList(listContainer, source, scheduler);
+    var list = new ScheduledList(listContainer, source);
+    var scheduler = ScheduledList.scheduler;
```
Comment (etiennesegonzac) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js]: we don't need to update this file right?

File: examples/bower_components/dom-scheduler/bower_components/fast-list/lib/scheduled-list.js [hunk 6]
```diff
@@ -4,10 +4,42 @@
   var startEvent = ('ontouchstart' in window) ? 'touchstart' : 'mousedown';
   var moveEvent = ('ontouchstart' in window) ? 'touchmove' : 'mousemove';
   var endEvent = ('ontouchstart' in window) ? 'touchend' : 'mouseup';
```
Comment (etiennesegonzac) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/lib/scheduled-list.js:6]: and this one

File: examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js (hunk 5 above)
Comment (wilsonpage) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js]: We do if dom-scheduler isn't loaded as this file must use the shim too.

File: test/karma.conf [hunk 7]
```diff
@@ -13,7 +13,7 @@ module.exports = function(config) {
 
     // list of files / patterns to load in the browser
     files: [
-      'lib/dom-scheduler.js',
+      'scheduled-list.js',
```
Comment (etiennesegonzac) [inline test/karma.conf]: and we should remove this change for now, will do the change as part of #16

**Comment by etiennesegonzac** [comment]
@wilsonpage last few comments then we're good to go!

File: examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js (hunk 5 above)
Comment (etiennesegonzac) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js]: when do we load this file as part of this repo ?
(none of the example app uses it I think)

File: examples/bower_components/dom-scheduler/bower_components/fast-list/lib/scheduled-list.js (hunk 6 above)
Comment (wilsonpage) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/lib/scheduled-list.js:6]: Hmm something strange going on with git here.

File: examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js (hunk 5 above)
Comment (wilsonpage) [inline examples/bower_components/dom-scheduler/bower_components/fast-list/js/page.js]: Oh I see what you mean now

File: examples/sections/index.html [hunk 8]
```diff
@@ -8,7 +8,7 @@
     <link rel="stylesheet" href="css/ul.css">
     <link rel="stylesheet" href="css/app.css">
 
-    <script src="bower_components/dom-scheduler/lib/dom-scheduler.js" defer></script>
+    <!-- <script src="bower_components/dom-scheduler/lib/dom-scheduler.js" defer></script> -->
```
Comment (wilsonpage) [inline examples/sections/index.html]: @etiennesegonzac do we want the example with or without dom-scheduler in the document?

Comment (etiennesegonzac) [inline examples/sections/index.html]: not sure, maybe the section with the scheduler and the simple one without?
(both with an easy comment to toggle on/off)

**Comment by wilsonpage** [comment]
@etiennesegonzac should all be addressed

**Comment by etiennesegonzac** [comment]
looks like `examples/simple/app.js` needs updating :)

**Comment by wilsonpage** [comment]
@etiennesegonzac done

**Comment by etiennesegonzac** [comment]
Looks like we're also missing the `bower_components` symlink in `examples/section`

**Comment by etiennesegonzac** [comment]
\o/

**Pull request merged**